src/vm/collection.rs

```diff
@@ -16,9 +15,12 @@ pub trait Collection<VM: VMBinding> {
     /// This method is called by a single thread in MMTk (the GC controller).
     /// This method should not return until all the threads are yielded.
     /// The actual thread synchronization mechanism is up to the VM, and MMTk does not make assumptions on that.
+    /// MMTk provide a callback function and expects the binding to use the callback for each mutator when it
+    /// is ready for stack scanning. Usually a stack can be scanned as soon as the thread stops in the yieldpoint.
     ///
     /// Arguments:
     /// * `tls`: The thread pointer for the GC controller/coordinator.
+    /// * `mutator_visitor`: A callback for each mutator to notify MMTk when a mutator is ready to be stack scanned.
```

Comment:
In English, "... for x to y" means "we do ... so that x can do y", and x becomes the subject.  This comment means "a callback so that each mutator can notify MMTk that it is ready".  But it may not always be the mutator that calls this callback (That would require `F` to implement `Send` and/or `Sync`).  It can be the current GC thread, too.  The GC thread can wait for some condition variable and notify MMTk on behalf of the mutator.

```suggestion
    /// * `mutator_visitor`: A callback.  Call it with a mutator as argument to notify MMTk that the mutator is ready to be scanned.
```

Reply:
Done.